src/compiler/compile/css/Selector.ts

```diff
@@ -470,10 +470,38 @@ function get_element_parent(node: Element): Element | null {
 	return parent as Element | null;
 }
 
+function find_previous_sibling_considering_slots(node: INode): INode {
```

Comment:
I might just call this `find_previous_sibling` and then add a JSDoc that describes how it treats slots